himin/day10/다트게임.js

```diff
@@ -0,0 +1,35 @@
+function solution(dartResult) {
+    const turns = dartResult.match(/\d{1,2}[SDT][*#]?/g); // 회차별로 분리
+    const scores = []; // 점수를 저장할 배열
+  
+    for (const turn of turns) {
+      let score = 0;
+  
+      // 숫자, 보너스, 옵션 분리
+      const match = turn.match(/(\d{1,2})([SDT])([*#]?)/);
```

Comment:
```suggestion
    const regex = /\d{1,2}[SDT][*#]?/g
    const turns = dartResult.match(regex); // 회차별로 분리
    const scores = []; // 점수를 저장할 배열
  
    for (const turn of turns) {
      let score = 0;
  
      // 숫자, 보너스, 옵션 분리
      const match = turn.match(regex);
```
정규표현식도 변수 처리가 가능합니다. 같은 내용이 여러번 반복되면, 변수로 처리하는게 좋아 보입니다.